### ISSUE TITLE
With Hardcore set to false normal survival is forced

### ISSUE THREAD

**Issue by Insane96** [GoryMoon/GlobalGameRules#26]
I want players to be able to choose survival or hardcore but if hardcore = false survival is forced.

This happens due https://github.com/GoryMoon/GlobalGameRules/blob/main/src/main/java/se/gory_moon/globalgamerules/WorldEvents.java#L78 enforcing the defined config hardcore if it's different from the current one. Should be changed to 
```java
if (hardcore && info.isHardcore() != hardcore)
```

**Comment by Krampus69** [comment]
Yes this has been annoying people who use Global in modpack (so a lot of people including people that play mine)

**Comment by Apollonu** [comment]
I know this issue has been closed for awhile, but this version is not on cf or modrinth and when I compiled it I still had the issue. Is this fix going to be posted on one or both of those platforms? I have difficulty lock on if it helps but I do not think it is causing any issues.
